test/functional/gridfs_stream.test.js

```diff
@@ -1422,4 +1422,25 @@ describe('GridFS Stream', function() {
       });
     });
   });
+
+  it('should correctly handle indexes create with exponent', function(done) {
+    const configuration = this.configuration;
+    const client = configuration.newClient();
+    client.connect((err, client) => {
+      expect(err).to.not.exist;
+      const db = client.db(configuration.db);
+      const col = db.collection('fs.files');
+      // prettier-ignore
+      col.createIndex({ filename: 1e0, uploadDate: 1e0 }, err => {
```

Comment:
I don't think there's a purpose to this in Javascript. Putting `1`, `1.0`, or even `1.0000000000000000001` here will all get converted to the same internal representation, if my understanding is correct.

Reply:
@emadum Spoke with @mbroadst yesterday and this was the direction we were going to go in.

Reply:
K, I still don't see the purpose but I'll go ahead and approve anyway.